### ISSUE TITLE
Multiple Issues

### ISSUE THREAD

**Issue by mrm-196** [yukimasano/self-label#3]
Hi

Trying to run the code, I came across the following issues which might be helpful if you fix them:
- Resnet architecture is incorrectly written as "presnet" in scripts/resnet.sh
- In return_model_loader function at data.py, the trainloader is only available for resnet architecture.
- Using only one GPU will result in division by zero multigpu.py
- Using more than one classification head leads to IndexError in CrossEntropy.
Could you please check these issues?

Thank you

**Comment by yukimasano** [comment]
Hi there, 

Thank you for the feedback, I've revamped the whole code and all the above issues should be solved. The only issue that I could not reproduce was the IndexError, so if that still persists, please let me know. Apart from that I've also added the CIFAR code and the evaluation codes.
Best, Yuki

**Comment by yukimasano** [comment]
ok, I've found the bug you mentioned. it's fixed now, so I'm closing it.

**Comment by mrm-196** [comment]
Hi,

Thanks for the fixes. The code works well now. However one minor issue that I'm still facing is when I set the number of heads to one, I get ValueError. Could you please check this as well?

Best,
Mohammadreza

**Comment by yukimasano** [comment]
Hi, I've tested both the cpu and the gpu version with number of heads = 1 and for me they seem to be working fine. Could you tell me with what settings you're getting the error?

**Comment by mrm-196** [comment]
Hi, this is the setting that I used:
```
device="0"
DIR="~/data/tiny-imagenet-200"

# the network
ARCH="alexnet"
HC=1
NCL=3000

# the training
WORKERS=4
BS=8
NEP=10
AUG=3

# the pseudo opt
NOPT=100

folder=pseudo${NCL}_${ARCH}_bs${BS}_hc${HC}-${NEP}_nopt${NOPT}_augT${AUG}

EXP=./${folder}
mkdir -p ${EXP}/checkpoints/L

python3 main.py \
        --cpu \
        --device ${device} \
        --imagenet-path ${DIR} \
        --exp ${EXP} \
        --batch-size ${BS} \
        --augs ${AUG} \
        --epochs ${NEP} \
        --nopts ${NOPT} \
        --hc ${HC} \
        --arch ${ARCH} \
        --ncl ${NCL} \
        --workers ${WORKERS} \
        --comment ${EXP}  | tee -a ${EXP}/log.txt;
```

**Comment by yukimasano** [comment]
Hi, I still couldn't reproduce the error. Can you please try with the newest changes in the repo (just pull again)? I would probably also suggest using a larger batch-size, e.g. using 128 or 256 should work fine for an alexnet and a 12GB GPU card. 

Good luck!
